core/src/main/java/org/openstack4j/openstack/networking/domain/ext/NeutronListenerV2.java

```diff
@@ -0,0 +1,280 @@
+import com.fasterxml.jackson.annotation.JsonRootName;
+import com.google.common.base.Objects;
+import org.openstack4j.model.network.ext.ListenerV2;
+import org.openstack4j.model.network.ext.Protocol;
+import org.openstack4j.model.network.ext.builder.ListenerV2Builder;
+import org.openstack4j.openstack.common.ListResult;
+
+import java.util.List;
+
+/**
+ * lbaas v2 listener
+ * @author emjburns
+ */
+@JsonRootName("listener")
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class NeutronListenerV2 implements ListenerV2 {
+    private static final long serialVersionUID = 1L;
+
+    private String id;
+
+    @JsonProperty("tenant_id")
+    private String tenantId;
+
+    private String name;
+
+    private String description;
+
+    /**
+     * The protocol of the VIP address. A valid value is TCP, HTTP, or HTTPS
+     */
+    private Protocol protocol;
+
+    /**
+     * The port on which to listen to client traffic that is associated with the
+     * VIP address. A valid value is from 0 to 65535.
+     */
+    @JsonProperty("protocol_port")
+    private Integer protocolPort;
+
+    /**
+     * The maximum number of connections allowed for the listener. Default is -1, meaning no limit.
+     */
+    @JsonProperty("connection_limit")
+    private Integer connectionLimit;
+
+    @JsonProperty("default_pool_id")
+    private String defaultPoolId;
+
+    @JsonProperty("loadbalancer_id")
+    private String loadbalancerId;
+
+    private List<ListItem> loadbalancers;
+
+    @JsonProperty("admin_state_up")
+    private boolean adminStateUp;
+
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<ListItem> getLoadBalancers(){
+        return loadbalancers;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isAdminStateUp(){
+        return adminStateUp;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getDefaultPoolId(){
+        return defaultPoolId;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Integer getConnectionLimit(){
+        return connectionLimit;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Integer getProtocolPort(){
+        return protocolPort;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Protocol getProtocol(){
+        return protocol;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getDescription(){
+        return description;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getName(){
+        return name;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getTenantId(){
+        return tenantId;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getId(){
+        return id;
+    }
+
+    @Override
+    public String toString(){
+        return Objects.toStringHelper(this)
```

Comment:
Suggest using either Lombok (mentioned above) or this method for all toString methods instead of manual String generation.